nni/retiarii/nn/pytorch/api.py

```diff
@@ -78,16 +78,21 @@ def __init__(self, candidates: Union[Dict[str, nn.Module], List[nn.Module]], lab
         self._label = generate_new_label(label)
 
         self.names = []
+        self.chosen = None
         if isinstance(candidates, OrderedDict):
             for name, module in candidates.items():
                 assert name not in ["length", "reduction", "return_mask", "_key", "key", "names"], \
                     "Please don't use a reserved name '{}' for your module.".format(name)
                 self.add_module(name, module)
                 self.names.append(name)
+                if not self.chosen:
+                    self.chosen = module
         elif isinstance(candidates, list):
             for i, module in enumerate(candidates):
                 self.add_module(str(i), module)
                 self.names.append(str(i))
+                if not self.chosen:
```

Comment:
Why do we need a chosen? This seems a `fix` mode and should be done in `__new__`.

Reply:
I need to call the first candidate. But as `candidates` is a list/dict, instead of `ModuleList`, so it can't be directly accessed in `forward` (eg., `candidates[0]`). I don't know if there are better ways to achieve that.

Reply:
In that case, you can write: `self._modules[self.names[0]](x)`

Reply:
> In that case, you can write: self._modules[self.names[0]](x)
Thanks, I got it.

Reply:
> In that case, you can write: `self._modules[self.names[0]](x)`

It seems I can't access `self._modules` in `forward`.

Reply:
In that case, you can write `self._first_module = self._modules[self.names[0]](x)`. Might be more clear that `self.chosen`.
My previous concern is that some one-shot algorithms use layer choice directly and might treat `self.chosen` as another module due to wrong implementation. But I found that actually `self.names` is used in `__iter__`. So never mind...